### PR TITLE
feat: Idp 887 refactor notify auth payment queue

### PR DESCRIPTION
**Pull request by sbellezz**
… classes and properties, changed queue for authPayment notification and updated test classes

### CODE REVIEW

File: helm/values.yaml [hunk 1]
```diff
@@ -63,19 +63,18 @@ microservice-chart:
     idpay-eventhub-00:
       KAFKA_BINDER_BROKER_ENDPOINT_LIST: kafka_broker
       KAFKA_TOPIC_NOTIFICATION: idpay_notification_request_topic
+
```
Comment (antonioT90) [inline helm/values.yaml:66]: remove also KAFKA_TOPIC_NOTIFICATION configuration: it should not more used (maybe you could remove the entire idpay-eventhub-00 block...)

**Review by antonioT90 — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/payment/service/qrcode/QRCodeAuthPaymentServiceImpl.java [hunk 2]
```diff
@@ -64,10 +60,16 @@ public AuthPaymentDTO authPayment(String userId, String trxCode) {
...
+        trx.setRejectionReasons(authPaymentDTO.getRejectionReasons());
+        sendAuthPaymentNotification(trx);
       } else {
         transactionInProgressRepository.updateTrxRejected(trx.getId(), authPaymentDTO.getRejectionReasons());
-        sendAuthPaymentNotification(trx, authPaymentDTO);
+        trx.setStatus(SyncTrxStatus.REJECTED);
+        trx.setReward(0L);
+        trx.setRejectionReasons(authPaymentDTO.getRejectionReasons());
+        sendAuthPaymentNotification(trx);
       }
```
[the remark this answers is not in the document]
Comment (sbellezz) [reply src/main/java/it/gov/pagopa/payment/service/qrcode/QRCodeAuthPaymentServiceImpl.java:74]: i need the trx updated before its passed to the _sendAuthPaymentNotification_ method

File: helm/values.yaml [hunk 3]
```diff
@@ -63,19 +63,18 @@ microservice-chart:
...
     idpay-eventhub-01:
       KAFKA_BROKER: kafka_broker
       KAFKA_ERRORS_BROKER: kafka_broker
       KAFKA_ERRORS_TOPIC: idpay_error_topic
-
-    idpay-eventhub-01:
       KAFKA_TRANSACTION_OUTCOME_BROKER: kafka_broker
       KAFKA_TRANSACTION_OUTCOME_CONFIG_SASL_MECHANISM: kafka_sasl_mechanism
       KAFKA_TRANSACTION_OUTCOME_CONFIG_SECURITY_PROTOCOL: kafka_security_protocol
```
Comment (sbellezz) [inline helm/values.yaml:73]: should i also remove these 2 properties since they are not used?

**Review by antonioT90 — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/payment/service/TransactionNotifierServiceImpl.java [hunk 4]
```diff
@@ -34,10 +34,13 @@ public Supplier<Flux<Message<Object>>> transactionOutcome() {
...
+    public boolean notify(TransactionInProgress trx, String headerValue) {
+       return streamBridge.send("transactionOutcome-out-0", binder, buildMessage(trx, headerValue));
     }
-    public static Message<TransactionInProgress> buildMessage(TransactionInProgress trx){
-        return MessageBuilder.withPayload(trx).setHeader(KafkaHeaders.KEY,trx.getMerchantId()).build();
+
+    public static Message<TransactionInProgress> buildMessage(TransactionInProgress trx, String headerValue) {
+        return MessageBuilder.withPayload(trx)
+                .setHeader(KafkaHeaders.KEY, headerValue)
+                .build();
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payment/service/TransactionNotifierServiceImpl.java:44]:
```suggestion
    public static Message<TransactionInProgress> buildMessage(TransactionInProgress trx, String key) {
        return MessageBuilder.withPayload(trx)
                .setHeader(KafkaHeaders.KEY, key)
                .build();
```

**Review by antonioT90 — APPROVED** [review]

**Pull request merged**